modules/programs/ne.nix

```
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.programs.ne;
+
+  autoPrefFiles = let
+    autoprefs = if cfg.defaultPreferences != "" then
+      cfg.automaticPreferences // {
+        ".default" = cfg.defaultPreferences;
+      } # include defaultPreferences as .default#ap
+    else
+      cfg.automaticPreferences;
+    keys = builtins.attrNames autoprefs;
+    gen = key: {
+      name = ".ne/${key}#ap"; # file path
+      value = {
+        text = builtins.getAttr key autoprefs;
+      }; # content, in the expected "path".text format
+    };
+  in listToAttrs (map gen keys); # assembling these into a regular record
+in {
+  meta.maintainers = [ hm.maintainers.cwyc ];
+
+  options.programs.ne = {
+    enable = mkEnableOption "ne";
+
+    keybindings = mkOption {
+      type = types.lines;
+      default = "";
+      example = ''
+        KEY 7f BS
+        SEQ "\x1b[1;5D" 7f
+      '';
+      description = ''
+        Keybinding file for ne
+      '';
+    };
+    defaultPreferences = mkOption {
```

Please separate the options with an empty line, it makes it easier to follow where they start and end. For example

```suggestion

    defaultPreferences = mkOption {
```